src/VirtoCommerce.CatalogPersonalizationModule.Data/Common/ModuleConfigurator.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hangfire;
+using VirtoCommerce.CatalogPersonalizationModule.Data.BackgroundJobs;
+using VirtoCommerce.CatalogPersonalizationModule.Data.Search.Indexing;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Settings;
+using VirtoCommerce.SearchModule.Core.Model;
+using static VirtoCommerce.CatalogPersonalizationModule.Core.ModuleConstants;
+
+namespace VirtoCommerce.CatalogPersonalizationModule.Data.Common
+{
+    public class ModuleConfigurator
+    {
+        private readonly ISettingsManager _settingsManager;
+        private readonly IEnumerable<IndexDocumentConfiguration> _documentIndexingConfigurations;
+        private readonly TaggedItemIndexChangesProvider _changesProvider;
+        private readonly CategoryTaggedItemDocumentBuilder _categoryDocumentBuilder;
+        private readonly ProductTaggedItemDocumentBuilder _productDocumentBuilder;
+
+        public ModuleConfigurator(ISettingsManager settingsManager, IEnumerable<IndexDocumentConfiguration> documentIndexingConfigurations, TaggedItemIndexChangesProvider changesProvider, CategoryTaggedItemDocumentBuilder categoryDocumentBuilder, ProductTaggedItemDocumentBuilder productDocumentBuilder)
+        {
+            _settingsManager = settingsManager;
+            _documentIndexingConfigurations = documentIndexingConfigurations;
+            _changesProvider = changesProvider;
+            _categoryDocumentBuilder = categoryDocumentBuilder;
+            _productDocumentBuilder = productDocumentBuilder;
+        }
+
+        public async Task ConfigureOutlinesSynchronizationJob()
```

Comment:
It seems there is the similar code in changes for each module.
I think we should write it once in platform.